dev/sg/sg_setup.go

```
@@ -286,11 +313,11 @@ programming languages and tools. Find out how to install asdf here:
 	https://asdf-vm.com/guide/getting-started.html
 
 Once you have asdf, execute the commands below.`,
-				instructionsCommands: `
+				instructionsCommands: stringer(`
 asdf plugin add nodejs https://github.com/asdf-vm/asdf-nodejs.git 
-echo 'legacy_version_file = yes' >> ~/.asdfrc
+touch ~/.asdfrc && cat ~/.asdfrc | grep "legacy_version_file = yes" || echo 'legacy_version_file = yes' >> ~/.asdfrc
```

There is a nasty bug here that I can reproduce in my mac VM but not on my local machine. Appending twice the configuration will lead `asdf` to ignore the legacy settings.

I'd curious to know if someone can reproduce it. Just run the echo part twice and run `asdf install nodejs` it should say that no version is specified and fail.

---

cat will return `1` if the file does not exist, which will break the evaluation.

---

Yeah, but that'll never happen since you create the file with `touch`, no? What am I missing? 😄 

```
$ touch does-not-exist2 && cat does-not-exist2 && echo 'exists'
exists
```

---

Ah, in that case it means we're ok with the error being printed, which is not the case with the touch. I assumed it would be better to do so. 

If you believe that's fine to have the error being printed, let's simplify it by using just `grep`.

---

I still don't understand - why not silence `grep`? Either redirect output to `/dev/null` or use `-s` to silence errors

```
~ φ rm -rf does-not-exist
~ φ grep -s 'foo' does-not-exist >/dev/null || (echo 'here is foo' >> does-not-exist && echo 'appended')
appended
~ φ grep -s 'foo' does-not-exist >/dev/null || (echo 'here is foo' >> does-not-exist && echo 'appended')
~ φ grep -s 'foo' does-not-exist >/dev/null || (echo 'here is foo' >> does-not-exist && echo 'appended')
```

---

It was just another way of silencing it while normalizing the user's config which isn't really required anyway; there isn't anything else to understand.